packages/mdc-drawer/modal/foundation.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCDrawerAdapter from '../adapter';
+import MDCDismissibleDrawerFoundation from '../dismissible/foundation';
+
+/**
+ * @extends {MDCDismissibleDrawerFoundation<!MDCDrawerAdapter>}
+ */
+class MDCModalDrawerFoundation extends MDCDismissibleDrawerFoundation {
+  static get defaultAdapter() {
```

Comment:
Shouldn't we update `MDCDismissibleDrawerFoundation`'s `defaultAdapter` to include everything and then we don't need this? (That's what e.g. top app bar does.)

It doesn't really make sense to have partial default adapters when there's only one actual adapter interface which includes everything. If there are variants that don't make use of everything, we tend to include guards within the adapter implementations to ensure they no-op rather than cause an error (e.g. in text field we'll check for existence of the line ripple or notched outline components).